context.go

```diff
@@ -102,36 +102,47 @@ func (context *Context) run(obj *wafObject, timeout time.Duration, cgoRefs *cgoR
 	ret := wafLib.wafRun(context.cContext, obj, result, uint64(timeout/time.Microsecond))
 
 	context.totalRuntimeNs.Add(result.total_runtime)
-	matches, actions, err := unwrapWafResult(ret, result)
+	events, actions, derivatives, err := unwrapWafResult(ret, result)
 	if err == ErrTimeout {
 		context.timeoutCount.Inc()
 	}
```

Comment:
You could definitively do a `defer` out of this and directly `return unwrapWafResult(...)` if `unwrapWafResult` were to return `Result`. Just a suggestion tho

Reply:
Is there a benefit I'm not seeing? It doesn't make the code easier to read and doesn't seem to solve a problem either.
Looks like using `defer` for the sake of it.

Reply:
I could argue about the happy path and some more stuff but as I said it's just a suggestion anyway. I am closing it